### PR TITLE
Project 19 - Caroline Fahlgren

### PR DESCRIPTION
**Pull request by CarolineF3**
Frontend: https://happy-thoughts-carolinef.netlify.app/
Backend: https://carolinef-happy-thoughts-api.herokuapp.com/

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,30 +1,90 @@
...
+app.use(cors());
+app.use(express.json());
 
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+app.get("/", (req, res) => {
+  res.json(listEndpoints(app));
+});
```
Comment (Ann-Sofi) [inline server.js:46]: Nice that you used list endpoints 👍

File: server.js [hunk 2]
```diff
@@ -1,30 +1,90 @@
...
+app.post("/thoughts/:id/like", async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const updatedThought = await Thought.findByIdAndUpdate(id, {
+      $inc: { hearts: 1 },
+    });
+    res.json(updatedThought);
+  } catch (error) {
+    res.status(404).json({ message: "Couldn't find thought by id" });
```
Comment (Ann-Sofi) [inline server.js:82]: Good, descriptive error message :-)

File: server.js [hunk 3]
```diff
@@ -1,30 +1,90 @@
...
+});
+
+app.get("/thoughts", async (req, res) => {
+  const { page = 0 } = req.query;
+
+  try {
+    const thoughts = await Thought.find()
+      .skip(page * 20)
+      .limit(20)
+      .sort({ createdAt: -1 });
```
Comment (Ann-Sofi) [inline server.js:55]: Easy to read when you break a line 🤩

**Review by Ann-Sofi — APPROVED** [review]
Good structure and nice, clean code! 
Like that you made a list of the endpoints in the readme.
Great work! ✨

File: server.js [hunk 4]
```diff
@@ -1,30 +1,90 @@
...
+
+const port = process.env.PORT || 8080;
+const app = express();
+
+const thoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    required: [true, "Message is required!"],
+    minlength: 5,
+    maxlength: 140,
```
Comment (KarolCislowski) [inline server.js:25]: Nicely configured field 👍🏻

File: server.js (hunk 3 above)
Comment (KarolCislowski) [inline server.js:55]: 👍🏻

File: server.js [hunk 5]
```diff
@@ -1,30 +1,90 @@
...
+    res.status(400).json({ message: "Couldn't save message" });
+  }
+});
+
+app.post("/thoughts/:id/like", async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const updatedThought = await Thought.findByIdAndUpdate(id, {
+      $inc: { hearts: 1 },
```
Comment (KarolCislowski) [inline server.js:78]: 💯

**Review by KarolCislowski — APPROVED** [review]
Great API 👍🏻

**Pull request closed without merging**